handlers/ecs/textextraction/s3handler.py

```diff
@@ -0,0 +1,188 @@
+# SOURCE: https://github.com/django/django/blob/master/django/core/files/storage.py
+# SOURCE: https://github.com/jschneier/django-storages/blob/master/storages/backends/s3boto3.py
+# NOTE: Only copied used part.
+
+import io
+import os
+import secrets
+import boto3
+import posixpath
+import mimetypes
+from gzip import GzipFile
+from tempfile import SpooledTemporaryFile
+from botocore.exceptions import ClientError
+
+
+def get_random_string(length, allowed_chars=(
+    'abcdefghijklmnopqrstuvwxyz'
+    'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'
+)):
+    """
+    SOURCE: DJANGO
+    Return a securely generated random string.
+    The bit length of the returned value can be calculated with the formula:
+        log_2(len(allowed_chars)^length)
+    For example, with default `allowed_chars` (26+26+10), this gives:
+      * length: 12, bit length =~ 71 bits
+      * length: 22, bit length =~ 131 bits
+    """
+    return ''.join(secrets.choice(allowed_chars) for i in range(length))
+
+def force_bytes(s):
+    if isinstance(s, bytes):
+        return s
+    if isinstance(s, memoryview):
+        return bytes(s)
+    return str(s).encode('utf-8', 'strict')
+
+
+def safe_join(base, *paths):
+    base_path = base
+    base_path = base_path.rstrip('/')
+    paths = [p for p in paths]
+    final_path = base_path + '/'
+    for path in paths:
+        _final_path = posixpath.normpath(posixpath.join(final_path, path))
+        # posixpath.normpath() strips the trailing /. Add it back.
+        if path.endswith('/') or _final_path + '/' == final_path:
+            _final_path += '/'
+        final_path = _final_path
```

Comment:
I am not entirely sure, but can we not just use pathlib Path? I think it might be cleaner and easier. We don't have to check for trailing slashes. More here: https://docs.python.org/3/library/pathlib.html
```python
from pathlib import Path

p1 = Path('/home/')
p2 = Path('bibek/')
print(p1 / p2)
# >> PosixPath("/home/bibek")
```

Reply:
let's keep it like that.